### PR TITLE
Allows opening a commissioning window on a bridged node 

### LINKED ISSUE #33798: [Feature] Allows opening a commissioning window on a bridged node that is not on Endpoint 0.
### Feature description

...
FabricSync spec changes were merged,
https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/multi_admin/fabric_synchronization.adoc

With the spec update, AdministratorCommissioningCluster is not always on endpoint 0, to open Commissioning Window on bridged node, we need to specify the endpointID

### Platform

all

### Platform Version(s)

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by yufengwangca**
AdministratorCommissioningCluster is not always on endpoint 0, to open Commissioning Window on bridged node, we need to support sending an OpenCommissioningWindow command with the given parameters to the given endpoint of the given node.

Fixes #33798

### CODE REVIEW

File: src/controller/CommissioningWindowOpener.h [hunk 1]
```diff
@@ -107,6 +107,39 @@ class CommissioningWindowOpener
...
+     * @param[in] callback      The function to be called on success or failure of opening of commissioning window.
+     * @param[out] payload      The setup payload, not including the VID/PID bits,
+     *                          even if those were asked for, that is generated
+     *                          based on the passed-in information.  The payload
+     *                          provided to the callback function, unlike this
+     *                          out parameter, will include the VID/PID bits if
+     *                          readVIDPIDAttributes is true.
+     *
+     */
+    CHIP_ERROR OpenCommissioningWindow(NodeId deviceId, EndpointId endpointId, System::Clock::Seconds16 timeout, uint32_t iteration,
```
Comment (andy31415) [inline src/controller/CommissioningWindowOpener.h:139]: These methods start taking too many arguments. Could you create some 'Params' class instead? Methods with more than 3-4 arguments are really hard to follow. This has 10. Same for below.

File: src/controller/CommissioningWindowOpener.h [hunk 2]
```diff
@@ -107,6 +107,39 @@ class CommissioningWindowOpener
                                        Callback::Callback<OnOpenCommissioningWindow> * callback, SetupPayload & payload,
                                        bool readVIDPIDAttributes = false);
 
+    /**
+     * @brief
+     *   Try to look up the bridged device on the bridge with the given node id
```
Comment (andy31415) [inline src/controller/CommissioningWindowOpener.h:112]: This class currently does not seem to reference bridges at all.

Could we re-phrase the documentation explaining that this is for use for bridges?

**Comment by andy31415** [comment]
@yufengwangca please update the PR summary with a description of changes. This seems to be "add ability in the CommissioningWindowOpener to specify the endpoint, so that bridges can be supported" and "enforce the existing opencommissioningwindow to check the endpoint id, since existing implementation is for the current device (i.e. handles root endpoint only)".

File: src/controller/CommissioningWindowOpener.h (hunk 2 above)
Comment (bzbarsky-apple) [inline src/controller/CommissioningWindowOpener.h:112]: None of this has anything to do with bridges, as far as this class is concerned.  The right thing to document is "send an OpenCommissioningWindow command with the given parameters to the given endpoint of the given node" or something along those lines.

The fact that this endpoint happens to be a bridge thing of some sort is irrelevant to this class.

File: src/controller/CommissioningWindowOpener.h [hunk 3]
```diff
@@ -107,6 +107,39 @@ class CommissioningWindowOpener
...
+     * @param[in] discriminator The long discriminator for the DNS-SD advertisement.
+     * @param[in] setupPIN      The setup PIN to use, or NullOptional to use a randomly-generated one.
+     * @param[in] salt          The salt to use, or NullOptional to use a
+     *                          randomly-generated one.  If provided, must be at
+     *                          least kSpake2p_Min_PBKDF_Salt_Length bytes and
+     *                          at most kSpake2p_Max_PBKDF_Salt_Length bytes in
+     *                          length.
+     * @param[in] callback      The function to be called on success or failure of opening of commissioning window.
+     * @param[out] payload      The setup payload, not including the VID/PID bits,
+     *                          even if those were asked for, that is generated
```
Comment (bzbarsky-apple) [inline src/controller/CommissioningWindowOpener.h:132]: What does "even if those were asked for" mean here?

Please watch the copy/paste....

File: src/controller/CommissioningWindowOpener.h [hunk 4]
```diff
@@ -137,6 +174,7 @@ class CommissioningWindowOpener
     Callback::Callback<OnOpenBasicCommissioningWindow> * mBasicCommissioningWindowCallback = nullptr;
     SetupPayload mSetupPayload;
     NodeId mNodeId                                       = kUndefinedNodeId;
+    EndpointId mEndpointId                               = kInvalidEndpointId;
```
Comment (bzbarsky-apple) [inline src/controller/CommissioningWindowOpener.h:177]:
```suggestion
    EndpointId mTargetEndpointId                               = kInvalidEndpointId;
```

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src/app/clusters/administrator-commissioning-server/administrator-commissioning-server.cpp [hunk 5]
```diff
@@ -106,6 +108,7 @@ bool emberAfAdministratorCommissioningClusterOpenCommissioningWindowCallback(
     VerifyOrExit(failSafeContext.IsFailSafeFullyDisarmed(), status.Emplace(StatusCode::kBusy));
 
     VerifyOrExit(!commissionMgr.IsCommissioningWindowOpen(), status.Emplace(StatusCode::kBusy));
+    VerifyOrExit(endpointId == kRootEndpointId, status.Emplace(StatusCode::kPAKEParameterError));
```
[the remark this answers is not in the document]
Comment (robszewczyk) [reply src/app/clusters/administrator-commissioning-server/administrator-commissioning-server.cpp:111]: This is the implementation of the proposed fabric sync feature.   For reference, see section [12.6.3.2. Aggregator Node] of the spec (the text came from https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/9106)

[the remark this answers is not in the document]
Comment (bzbarsky-apple) [reply src/app/clusters/administrator-commissioning-server/administrator-commissioning-server.cpp:111]: Nothing in that spec section says "Ignore command invokes on the Administrator Commissioning cluster if it's not on endpoint 0".

[the remark this answers is not in the document]
Comment (yufengwangca) [reply src/app/clusters/administrator-commissioning-server/administrator-commissioning-server.cpp:111]: Registered a ticket to clarify the intended behavior when receives OCW command not directed to Endpoint 0
 https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/9354

[the remark this answers is not in the document]
Comment (bzbarsky-apple) [reply src/app/clusters/administrator-commissioning-server/administrator-commissioning-server.cpp:111]: Great.  Until that's resolved, do not make any changes to this cluster impl, please, because those changes would make it current-spec-not-compliant.

**Comment by yufengwangca** [comment]
This PR is no more valid after https://github.com/project-chip/connectedhomeip/pull/34209 land, withdraw here and Abdul will create a new one
